### PR TITLE
Stateful wrappers [DelayObservation ,TimeAwareObservation, StickyAction]

### PR DESCRIPTION
**Pull request by gianlucadecola**
This PR add the following stateful wrappers:  

* `DelayObservationV0`: adds a delay to the returned observation.
* `StickyActionV0`: adds a probability of repeating the previous executed action.
* `TimeAwareObservationV0`: add time informations to the observation.  

Behaviour logic is mostly taken from the respective supersuit wrappers.

### TODOS:

- [x] docs

### CODE REVIEW

File: gymnasium/wrappers/time_aware_observation.py [hunk 1]
```diff
@@ -27,6 +28,10 @@ def __init__(self, env: gym.Env):
         Args:
             env: The environment to apply the wrapper
         """
+        deprecation(
+            "`TimeAwareObservation` is marked as deprecated and will be removed in the near future."
+            "Use TimeAwareObservationV0 instead which allows support for composite action space and Jax."
+        )
```
Comment (gianlucadecola) [inline gymnasium/wrappers/time_aware_observation.py:34]: Is this needed here or should I remove it for now?

File: gymnasium/dev_wrappers/time_aware_observation.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            The environment's step using the action.
+        """
+        self.timesteps += 1
+        observation, reward, terminated, truncated, info = super().step(action)
+
+        self.timesteps = jp.where(terminated, 0, self.timesteps)
+
+        return observation, reward, terminated, truncated, info
+
+    def reset(self, **kwargs):
```
Comment (pseudo-rnd-thoughts) [inline gymnasium/dev_wrappers/time_aware_observation.py:112]: Add arguments

File: gymnasium/dev_wrappers/time_aware_observation.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        return super().reset(**kwargs)
+
+    @staticmethod
+    def _get_max_timesteps(env: gym.Env):
+        """Get the max episode steps before truncation occurs.
+
+        Args:
+            env: the provided environment
+        """
+        return (
```
Comment (pseudo-rnd-thoughts) [inline gymnasium/dev_wrappers/time_aware_observation.py:131]: Remove, this should be able to be simplified to just consider `gym.Env` not `VectorEnv`

File: gymnasium/wrappers/time_aware_observation.py (hunk 1 above)
Comment (pseudo-rnd-thoughts) [reply gymnasium/wrappers/time_aware_observation.py:34]: I have changed how I think the development should work going forward https://github.com/Farama-Foundation/Gymnasium/pull/159

**Comment by pseudo-rnd-thoughts** [comment]
Similar to the lambda action PR, could you 
* Move the wrappers to `experimental.wrappers`
* Remove the deprecation warnings
* Update the docs

Im uncertain about the files, I would prefer to have less files but there is a level where that is not possible

File: gymnasium/wrappers/time_aware_observation.py (hunk 1 above)
Comment (gianlucadecola) [reply gymnasium/wrappers/time_aware_observation.py]: https://github.com/Farama-Foundation/Gymnasium/pull/165/commits/a6c22fa77aecf49af3829e2c591c5abb7c41e8cb :+1:

File: gymnasium/dev_wrappers/sticky_action.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        """Execute the action."""
+        if (
+            self.old_action is not None
+            and self.np_random.uniform() < self.repeat_action_probability
+        ):
+            action = self.old_action
+        self.old_action = action
+        return action
+
+    def reset(self, **kwargs):
```
[the remark this answers is not in the document]
Comment (gianlucadecola) [reply gymnasium/dev_wrappers/sticky_action.py]: You mean `def reset(self, *args, **kwargs)`?

File: gymnasium/dev_wrappers/time_aware_observation.py (hunk 3 above)
Comment (gianlucadecola) [reply gymnasium/dev_wrappers/time_aware_observation.py]: https://github.com/Farama-Foundation/Gymnasium/pull/165/commits/75d7978551caa9416c706e8c30c72888943b63ad

**Review by pseudo-rnd-thoughts — APPROVED** [review]

**Pull request merged**